src/pages/benchviz/Benchmark.tsx

```diff
@@ -0,0 +1,151 @@
+    const rowColor = change === 0 ? "currentColor" : change > 0 ? "red" : "green";
+
+    resultsTable.push(
+        <tr key={i} style={{ color: rowColor }}>
+            <td>{bm.left?.benchmarkName}</td>
+            <td>{formatMemory(benchmarkGarbage(bm.left!))}</td>
+            <td>{formatMemory(benchmarkGarbage(bm.right!))}</td>
+            <td>{change}</td>
+        </tr>,
+    );
+});
+
+// Comparison data master garbage created vs commit garbate created (ABSOLUTE)
+const generatedGarbageData = benchmarksSortedByPercentDifference.map((bm) => ({
+    name: formatBenchmarkName(bm.left?.benchmarkName ?? bm.right?.benchmarkName!),
+    oldValue: bm.left?.categories[MemoryBenchmarkCategory.Garbage] ?? 0,
+    newValue: bm.right?.categories[MemoryBenchmarkCategory.Garbage] ?? 0,
+}));
+
+let garbageCreatedComparisonSvg: SVGSVGElement;
+const garbageCreatedComparisonGraphWidth = 1000;
+const garbageCreatedComparisonGraphHeight = 300;
+
+// Comparison data master garbage created vs commit garbate created (ABSOLUTE)
+const generatedGarbageChangeData = matchedBenchmarks.map((bm) => {
+    const oldValue = bm.left?.categories[MemoryBenchmarkCategory.Garbage]!;
+    const newValue = bm.right?.categories[MemoryBenchmarkCategory.Garbage]!;
+
+    return {
+        name: formatBenchmarkName(bm.left?.benchmarkName ?? bm.right?.benchmarkName!),
+        value: (100 * (newValue - oldValue)) / oldValue,
+    };
+});
+
+let garbageCreatedChangeSvg: SVGSVGElement;
+const garbageCreatedChangeGraphWidth = 1000;
+const garbageCreatedChangeGraphHeight = 300;
+
+export default class extends React.Component {
+    componentDidMount() {
+        this.draw();
+    }
+
+    componentDidUpdate() {
+        this.draw();
+    }
+
+    draw() {
+        positiveNegativeBarGraph(
+            d3.select(garbageCreatedChangeSvg),
+            generatedGarbageChangeData,
+            garbageCreatedChangeGraphWidth,
+            garbageCreatedChangeGraphHeight,
+        );
+
+        barComparisonGraph(
+            d3.select(garbageCreatedComparisonSvg),
+            generatedGarbageData,
+            garbageCreatedComparisonGraphWidth,
+            garbageCreatedComparisonGraphHeight,
+        );
+    }
+
+    render() {
+        return (
+            <>
+                {/* Results table */}
+                <h2>Benchmark results</h2>
+                <table>
+                    <thead>
+                        <tr style={{ fontWeight: "bold" }}>
+                            <td>Benchmark</td>
+                            <td>Garbage Master</td>
+                            <td>Garbage Commit</td>
+                            <td>% Change</td>
+                        </tr>
+                    </thead>
+                    <tbody>{resultsTable}</tbody>
+                </table>
+
+                <h2>Garbage created change</h2>
+                {/* [% Delta] Gerbage created */}
+                <svg
+                    ref={(n) => (garbageCreatedChangeSvg = n!)}
```

Comment:
The conventional way to use refs now is `useRef` (or `createRef` in the case of class components)

```ts
export default function Benchmark() {
    const garbageCreatedChangeRef = useRef<SVGSVGElement>(null!);
    useEffect(() => {
        d3.select(garbageCreatedChangeRef.current);
    });

    return (
        // ...
        <svg ref={garbageCreatedChangeRef}
        // ...
    );
}
```